modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/TypeWrapperTestCase.java

```diff
@@ -63,6 +70,25 @@ public void filterInternalJdkFields() {
     assertThat(fields, empty());
   }
 
+  @Test
+  @Issue("W-13648907")
+  @Description("Ensure synthetic fields are filtered out as they might be added to classes we can't use reflection on in Java 17 and above")
+  public void filterSyntheticFields() {
+    // precondition check
+    final Collection<String> allFiledNames = IntrospectionUtils.getFields(SomeClass.class)
+        .stream()
+        .map(Field::getName)
+        .collect(toSet());
+    // being a non-static inner class makes it have a synthetic field pointing to its owner instance.
+    assertThat(allFiledNames, hasItem("this$0"));
```

Comment:
you should check by the same condition that the productive code is checking. Also this assertion would break is future java versions change this impl.